dev/io.openliberty.org.apache.myfaces.4.0_fat/fat/src/io/openliberty/org/apache/myfaces40/fat/tests/MultipleInputFileTest.java

```diff
@@ -0,0 +1,355 @@
+
+        Iterator<DomElement> iterator = messages.getChildElements().iterator();
+        assertEquals("Uploaded file has been received", "field: " + field.name + ", name: " + file.getName() + ", size: " + file.length(), iterator.next().asText());
+        assertFalse("Iterator should not have had another element", iterator.hasNext());
+    }
+
+    /**
+     * Asserts that a multiple file selection results in the correct messages output.
+     *
+     * @param webClient - The web client used to generate the page
+     * @param page - The html page
+     * @param usingAjax - true is using Ajax, false otherwise.
+     * @param field - is the form input using single, or multiple inputs
+     * @throws Exception - Thrown if we were not able to execute the assertion testing
+     */
+    private void assertMultipleSelection(HtmlPage page) throws Exception {
+        //Get the file input element
+        HtmlFileInput input = page.getHtmlElementById("form:input");
+        assertEquals("Multiple attribute is set", "multiple", input.getAttribute("multiple"));
+
+        //Create files and add them to the input attribute
+        File file1 = generateTempFile("file1", "bin", 123);
+        File file2 = generateTempFile("file2", "bin", 234);
+        File file3 = generateTempFile("file3", "bin", 345);
+        input.setValueAttribute(file1.getAbsolutePath());
+        addValueAttribute(input, file2.getAbsolutePath());
+        addValueAttribute(input, file3.getAbsolutePath());
+
+        //Click submit
+        page = page.getHtmlElementById("form:submit").click();
+        FATSuite.logOutputForDebugging(server, page.asXml(), name.getMethodName() + ".clicked.html");
+
+        //Assert messages are expected
+        HtmlElement messages = page.getHtmlElementById("messages");
+        assertEquals("There are 3 messages", 3, messages.getChildElementCount());
+
+        Iterator<DomElement> iterator = messages.getChildElements().iterator();
+        assertEquals("First uploaded file has been received", "field: multipleSelection, name: " + file1.getName() + ", size: " + file1.length(),
+                     iterator.next().asText());
+        assertEquals("Second uploaded file has been received", "field: multipleSelection, name: " + file2.getName() + ", size: " + file2.length(),
+                     iterator.next().asText());
+        assertEquals("Third uploaded file has been received", "field: multipleSelection, name: " + file3.getName() + ", size: " + file3.length(),
+                     iterator.next().asText());
+        assertFalse("Iterator should not have had another element", iterator.hasNext());
+    }
+
+    /**
+     * @param webClient
+     * @param page
+     * @param usingAjax
+     * @param field - is the form input using single, or multiple inputs
+     * @throws IOException
+     */
+    private void assertMultipleSelectionProducesSingleResult(HtmlPage page) throws IOException {
```

Comment:
I'm confused by the description? You only have one parameter, but document multiple?  Similar comment for assertMultipleSelection and assertSingleSelection.